lib/preload.js

```diff
@@ -237,26 +247,25 @@ const getLayerSize = (registryHost, token, imageRepo, blobSum) => {
   })
 }
 
-const getContainerSize = (apiHost, apiToken, registryHost, imageRepo) => {
-  const progress = new visuals.Progress(`Fetching container size for ${imageRepo}`)
-  progress.update({ percentage: 0 })
+const getContainerSize = (apiHost, apiToken, registryHost, imageRepo, fetchSizeProgress) => {
+  fetchSizeProgress.emit('progress', 0)
```

Comment:
How will I return the container size then?
I could do return { progress, promise } but that doesn't look like a good solution to me.

Reply:
Emitting it through an event, for example:
```js
const getContainerSize = (options) => {
  // ...
  // Once you've established what the container size is, just
  progress.emit('size', containerSize)
  // ...
  return progress
}
```
Generally, I think mixing Promises with EventEmitters or Streams is something to avoid, as they're different types of control flow handlers – I'd just pick one or the other, and be consistent.

Reply:
@jhermsmeier I just tried this approach, but it makes things very complicated for the caller (`run`) forcing it to return a Promise of an object (itself containing a promise and several event emitters).
I'd prefer to stick to the current approach.
I've changed `getContainerSize` to take a single `options` object argument